### ISSUE TITLE
Help needed: Stuck with a rogue firmware install..

### ISSUE THREAD

**Issue by mrishi** [Dasharo/dasharo-issues#492]
**Dasharo version**
`v1.1.1`

**Dasharo variant**
`MSI Desktops - MSI PRO Z690-A`

**Question text**
Hi,

I got  a bit excited to see that there is an open-source firmware for my **MSI Z690 PRO A WIFI D4** motherboard, which was giving me multiple issues with the proprietory BIOS from MSI, depsite the number of updates they released over the last year. Some of the issues I was facing was:
- Unable to set iGPU as the main graphics card when and dGPU is connected
- Messed up IOMMU groups
- ACPI errors due to duplicate assignments
- Many more..

So, I decided to take the plunge and installed the community version using DTS that was successful.

But that's where all my nightmares started. The BIOS wouldn't load and would enter into a boot loop every few seconds.  I tried button flashing method to recover original firmware, but that didn't work either.

Reading through the documentation, I found that the community version(1.1.1) I stalled:
- Does not support Intel Raptor Lake processors
- Breaks the MSI button flash method

I boorrwed a 12700 Alder Lake processor from a friend and replaced it with my 13600 Raptor Lake processor. The BIOS loaded successfully and I could boot into the OS (Windows 11) too. But I have no way to get my 13600 functional with my PC currently. 

I also figured out that the version 1.1.2 (Dasharo Entry subscription release) fixes both Raptor Lake support & MSI Button Flash Recovery. I have no issues going for the subscription if that help, but got no idea how I will flash the bios, as Dasharo has not option to flash the BIOS in BIOS mode ( like MSI).

Pls. help..

**Comment by zirblazer** [comment]
Do you remember if DTS did a backup of your original MSI Firmware in the DTS USB Flash Drive? You can restore it by manually flashing: https://docs.dasharo.com/dasharo-tools-suite/documentation/#local-firmware-update

sudo flashrom -p internal --ifd -i bios -w whatevername.rom

Do remember to disable the flash protections in Firmware before attemping, BIOS boot medium lock and Enable SMM BIOS write protection have to be off: https://docs.dasharo.com/dasharo-menu-docs/dasharo-system-features/

**Comment by mrishi** [comment]
Thanks for your response.

> Do you remember if DTS did a backup of your original MSI Firmware in the DTS USB Flash Drive? You can restore it by manually flashing: https://docs.dasharo.com/dasharo-tools-suite/documentation/#local-firmware-update
> 
> sudo flashrom -p internal --ifd -i bios -w whatevername.rom
> 
> Also remember to disable the flash protections in Firmware before attemping, BIOS boot medium lock and Enable SMM BIOS write protection, and Secure Boot have to be off: https://docs.dasharo.com/dasharo-menu-docs/dasharo-system-features/

It did take a backup which was sent(uploaded) to Dasharo backend as part of the HCL report successfully. But I did not find any files in the USB drive, so I formtated it later for other purpose.

Are you suggesting to load DTS using 12th Gen CPU? I don't have access to it, as I am back to my 13th Gen CPU which does not load the BIOS? Any other way to load it? PXE network boot or something..

> I also figured out that the version 1.1.2 (Dasharo Entry subscription release) fixes both Raptor Lake support & MSI FlashBIOS button Recovery/update. I have no issues going for the subscription if that helps, but got no idea how I will flash the bios as Dasharo has no option to flash the BIOS in BIOS mode(like MSI).

I managed to compile and build v1.1.2 on another machine, which apparently fixes these problems as mentioned in my post above. Do I still need the subscription for flashing the  binary, or I can go ahead and flash it? As per releases notes, it says this is a Dasharo Entry Subscription release.

**Comment by SergiiDmytruk** [comment]
> Do I still need the subscription for flashing the binary

No.  Lack of subscription doesn't affect functionality in any way.
